### PR TITLE
[cryo] Use mysql options file for setting password

### PR DESCRIPTION
**Pull request by wyao**
Reduce the risk of leaking the MySQL password.

@jtai

### CODE REVIEW

**Comment by wyao** [comment]
And @philsnow

File: lib/cryo/database/mysql.rb [hunk 1]
```diff
@@ -14,14 +14,45 @@ def initialize(opts={})
...
   def get_backup
-    safe_run "mysqldump --host=#{host} --user=#{user} --password=#{password} --all-databases --ignore-table=mysql.slow_log_backup --ignore-table=mysql.slow_log --single-transaction > #{local_path}"
+    write_cnf
+    begin
+      safe_run "mysqldump --defaults-file=#{cnf_path} --host=#{host} --user=#{user} --all-databases --ignore-table=mysql.slow_log_backup --ignore-table=mysql.slow_log --single-transaction > #{local_path}"
+    rescue => e
+      delete_cnf
+      throw e
+    end
+    delete_cnf
```
Comment (jtai) [inline lib/cryo/database/mysql.rb]: Why not `ensure`?

Comment (wyao) [inline lib/cryo/database/mysql.rb]: Because. N00b.

**Comment by wyao** [comment]
Thanks Phil. Your expertise is always appreciated :)

**Comment by wyao** [comment]
I will use `Tempfile` but still explicitly unlink the file.

> When a Tempfile object is garbage collected, or when the Ruby interpreter exits, its associated temporary file is automatically deleted. This means that’s it’s unnecessary to explicitly delete a Tempfile after use, though it’s good practice to do so: not explicitly deleting unused Tempfiles can potentially leave behind large amounts of tempfiles on the filesystem until they’re garbage collected. The existence of these temp files can make it harder to determine a new Tempfile filename.

**Comment by wyao** [comment]
Thanks Phil. Updated with your recommendations.

**Comment by philsnow** [comment]
:+1: 

is mysqldump clever enough to not need the final newline in its .cnf file ?

**Comment by wyao** [comment]
Yup! I tested this in the actual program.

**Pull request merged**